### ISSUE TITLE
Query Object Variable Error - GraphQL.ExecutionError: Expected non-null value

### ISSUE THREAD

**Issue by markchilton** [graphql-dotnet/graphql-dotnet#979]
Hey everyone.

I am trying to query GraphQL using a request object. However I keep getting this error when building the schema: 

> GraphQL.ExecutionError: Expected non-null value, resolve delegate return null for \"$BeSoftware.Services.Models.Requests.PlayerBalanceRequestType\" ---> System.InvalidOperationException: Expected non-null value, resolve delegate return null for \"$BeSoftware.Services.Models.Requests.PlayerBalanceRequestType\"\r\n

I have the query (below) in BeSoftwareQLQuery.cs

```
Field<PlayerBalanceType>(
 "playerBalance",
 arguments: new QueryArguments(new QueryArgument<NonNullGraphType<PlayerBalanceRequestType>> { Name = "request" }),
 resolve: context => playerService.GetPlayerBalanceAsync(context.GetArgument<PlayerBalanceRequest>("request")));
```

My PlayerBalanceRequestType class looks like this:

```    
public class PlayerBalanceRequestType : ObjectGraphType<PlayerBalanceRequest>
    {
        public PlayerBalanceRequestType()
        {
            Name = "request";
            Field(x => x.Id);
            Field(x => x.Login);
            Field(x => x.ForMoney);
        }
    }
```

and my PlayerBalanceRequest class looks like this:

```
  public class PlayerBalanceRequest
    {
        public int Id { get; set; }

        public string Login { get; set; }

        public bool ForMoney { get; set; }
    }
```

Could someone tell me where I am going wrong please?

Thanks
Mark

**Comment by joemcbride** [comment]
There are other issues with this same error. Can you search through those and see if the responses apply to you?  Thanks.

**Comment by markchilton** [comment]
Hi, I already did look through similar issues. The closest one I could find is this: 

https://github.com/graphql-dotnet/graphql-dotnet/issues/930

However I followed and applied the suggested fix but am still getting the same error.

**Comment by joemcbride** [comment]
PlayerBalanceRequestType should be an InputObjectGraphType.

**Comment by markchilton** [comment]
Thanks for the reply, I already tried that but it didn't fix the issue.

I changed my PlayerBalanceRequestType class to look like this:

```
    public class PlayerBalanceRequestType : InputObjectGraphType<PlayerBalanceRequest>
    {
        public PlayerBalanceRequestType()
        {
            Name = "request";
            Field(x => x.Id);
            Field(x => x.Login);
            Field(x => x.ForMoney);
        }
    }
```

**Comment by Tarig0** [comment]
Have you tried Async field? Looks like the issue may be you are returning Task<PlayerBalance> and not PlayerBalance

**Comment by markchilton** [comment]
That didn't make any difference unfortunately. GraphQL works just fine if I pass in all the types as separate QueryArguments. It just doesn't seem to like me passing in an object as a QueryArgument.

**Comment by Tarig0** [comment]
Ah I see, did you add the PlayerBalanceRequestType to the DI container

**Comment by OpenSpacesAndPlaces** [comment]
Pseudo like the following works for me:

```
InputObjectGraphType<MyObject> o = new InputObjectGraphType<MyObject>();
o.Name = "UniqueObjectName";
o.Field(FieldGraphType,"FieldName");
QueryArgument qa = new QueryArgument(o);
qa.Name = "data";
qa.Description = "UniqueObjectDesc";
TableArgs.Add(qa);
```

Duplicate names/lack of names/incorrect inferred graph types could all be culprits from what I've seen before.

I don't know if it would help you, but I've had good luck with the GraphTypes  explicitly defined vs. inferred with syntax sugar.

e.g.
```
AddField(new FieldType
                {
                    Name = "playerBalance",
                    Type = typeof(PlayerBalanceType),
                    ResolvedType = GraphType,
....
```

**Comment by sungam3r** [comment]
Fixed in #1462 .
